ui/src/app/components/runtime-configuration-panel/customize-panel.spec.tsx

```diff
@@ -383,4 +385,70 @@ describe(CustomizePanel.name, () => {
     expect(screen.queryByText(/Warning Number One/)).toBeInTheDocument();
     expect(screen.queryByText(/Warning Number Two/)).toBeInTheDocument();
   });
+
+  // many subcomponents are enabled/disabled by the same logic.  PresetSelector can act as a stand-in here for:
+  // MachineSelector, GpuConfigSelector, DataProcConfigSelector, the autopause Dropdown, DiskSelector, and
+  // Delete Environment in CustomizePanelFooter.
```

Comment:
I think I prefer that the original test simply refers to "controls." My (possibly naive) idea of a "best" test here would be one that asserts that every control is disabled instead of relying on a comment and knowledge of the underlying logic.

Reply:
I like that idea.  Unfortunately I don't have a great check for "is it disabled" besides "does clicking it do anything", and that has some subtleties for each of these components.